File: abc/abc127/abc127_c.cs
// https://atcoder.jp/contests/abc127/submissions/8380301

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        public static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = input[0];
            var M = input[1];

            var l = 1;
            var r = N;
            for (int i = 0; i < M; i++)
            {
                input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
                var L = input[0];
                var R = input[1];

                l = Math.Max(l, L);
                r = Math.Min(r, R);
            }
            Console.WriteLine(Math.Max(0, r - l + 1));
        }
    }
}

File: abc/abc127/abc127_c.fs
// https://atcoder.jp/contests/abc127/submissions/12233685

let [| N; M |] = stdin.ReadLine().Split() |> Array.map int

let rec solver l r i =
    if i = M then max 0 (r - l + 1)
    else
        let [| L; R |] = stdin.ReadLine().Split() |> Array.map int
        solver (max l L) (min r R) (i + 1)

solver 1 N 0 |> stdout.WriteLine
